File: src/main/java/ash/java/graphql/fields/TvSeasonSearchSchema.java
package ash.java.graphql.fields;


import ash.java.graphql.data.TvDao;
import ash.java.graphql.types.tvseason.TvSeasonType;
import graphql.schema.GraphQLFieldDefinition;
import graphql.schema.GraphQLNonNull;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import static graphql.Scalars.*;
import static graphql.schema.GraphQLFieldDefinition.newFieldDefinition;

@Service
public class TvSeasonSearchSchema implements FieldProducer {

    private TvDao tvDao;

    @Autowired
    public TvSeasonSearchSchema(TvDao tvDao) {
        this.tvDao = tvDao;
    }

    private GraphQLFieldDefinition tvSeasonSearchFieldDefinition = newFieldDefinition()
            .type(new TvSeasonType().getGraphQlType())
            .name("tvSeasonSearch")
            .argument(arg -> arg.name("tvId").type(new GraphQLNonNull(GraphQLInt)))
            .argument(arg -> arg.name("seasonNum").type(new GraphQLNonNull(GraphQLInt)))
            .dataFetcher(env -> tvDao.getTvSeason(env.getArgument("tvId"), env.getArgument("seasonNum")))
            .build();

    @Override
    public GraphQLFieldDefinition getFieldDefinition() {
        return tvSeasonSearchFieldDefinition;
    }
}

File: src/main/kotlin/ash/kotlin/graphql/fields/TvSeasonSearchSchema.kt
package ash.kotlin.graphql.fields

import ash.kotlin.graphql.data.TvDao
import ash.kotlin.graphql.types.tvseason.TvSeasonType
import graphql.Scalars
import graphql.schema.GraphQLFieldDefinition
import graphql.schema.GraphQLNonNull
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class TvSeasonSearchSchema @Autowired constructor(val dao: TvDao) : FieldProducer {
    override fun getFieldDefinition(): GraphQLFieldDefinition {
        return GraphQLFieldDefinition.newFieldDefinition()
                .type(TvSeasonType().getGraphQlType())
                .name("tvSeasonSearch")
                .argument { arg -> arg.name("tvId").type(GraphQLNonNull(Scalars.GraphQLInt)) }
                .argument { arg -> arg.name("seasonNum").type(GraphQLNonNull(Scalars.GraphQLInt)) }
                .dataFetcher { env -> dao.getTvSeason(env.getArgument("tvId"), env.getArgument("seasonNum"))}
                .build()
    }
}